V0.1

kaevinio opened the pull request:
Aktuell greift der SDK noch auf Roadkit Dev in Firebase zu, weil die Cloud Functions live noch nicht deployed sind.

kaevinio committed:
ROK-7: SDK erstellen
- Package erstellt
- Setup von ProjektID und UserID
- Laden vorhandener Topics anhand der ProjektID

kaevinio committed:
- Model für Topic angepasst
- Model für neues Topic erstellt
- Routen in separatem Struct hinzugefügt
- TopicStatus & TopicPriority hinzugefügt
- URLRequest erstellen mit HTTPMethod und optionalem Body
- Topic voten & erstellen

kaevinio committed:
- ObjectModel und ObjectViewModel für SwiftUI hinzugefügt
- "Improvement" als type hinzugefügt
- Parameter aus Objekten entfernt, die über Cloud Functions gesetzt werden
- Route zum Vorschlagen entfernt

annanna commented on Roadkit/Sources/Roadkit/API/Routes.swift:
```diff
@@ -0,0 +1,40 @@
+    
+    var projectID = ""
+    var topicID = ""
+    var userID = ""
```
warum gibt es hier einen Standardwert? Die sollte ja nie leer sein, oder?

kaevinio replied:
Jain, du hast Recht, ich die UserID nicht mehr optional aber projectID und topicID werden nicht bei beiden gebraucht

annanna commented on Roadkit/Sources/Roadkit/API/Routes.swift:
```diff
@@ -0,0 +1,40 @@
+        case topics, vote
+    }
+    
+    private let host = "https://europe-west3-roadkit-dev-swift.cloudfunctions.net/topics/"
```
das zeigt noch auf Dev

annanna replied:
(hast du geschrieben, ich weiß, aber die Cloud Functions sind ja approved)

annanna commented on Roadkit/Sources/Roadkit/Enums/HTTP/HTTPMethod.swift:
```diff
@@ -0,0 +1,14 @@
+import Foundation
+
+enum HTTPMethod: String {
+    case get = "GET",
```
könntest hier auch mit `rawValue.uppercased` arbeiten

annanna commented on Roadkit/Package.swift:
```diff
@@ -0,0 +1,29 @@
+// swift-tools-version: 5.5
```
ganz grundsätzlich ist hier alles eine Ebene zu hoch, also das `Package.swift`-File muss direkt in die Root, sonst kannst du es nicht über XCode hinzufügen

annanna commented on Roadkit/Sources/Roadkit/RoadkitManager.swift:
```diff
@@ -0,0 +1,178 @@
+     - Parameter projectId: The ID of your app
+     - Parameter userId: The ID of your currently logged in user (can be empty)
+     */
+    public func setupRoadkit(projectID: String, userID: String) {
```
Mh ja, die ist ja eigentlich nicht optional, oder? Also den Rest kann man ja nicht nutzen, wenn die `projectId` und userId leer sind?
Könnten es entweder wie bei Firebase machen, dass mit Absicht Crashes geworfen werden, wenn man vorher nicht `configure()` aufgerufen hat. Oder wir nehmen einfach ein init 😬

kaevinio replied:
Also die UserID kann leer sein (wenn man sich in einer App nicht anmelden muss) aber für die ProjectID wird jetzt gechecked und ansonsten crasht es.

annanna commented on Roadkit/Sources/Roadkit/API/NetworkResponse.swift:
```diff
@@ -0,0 +1,12 @@
+//
+//  File.swift
```
filename

kaevinio committed:
- Files eine Ebene nach oben verschoben
- Crash, wenn keine ProjectID bereitgestellt wurde
- rawValue für HTTPMethod ersetzt
- Cloud Functions zeigen auf Production

kaevinio committed:
- URL für Prod angepasst

kaevinio committed:
- Error werfen, wenn die UserID nicht gesetzt wurde
